doc/coq.rst

```diff
@@ -105,12 +105,20 @@ The semantics of the fields are:
 - Your Coq theory can depend on other theories by specifying them in the
   ``<coq_theories>`` field. Dune then passes to Coq the corresponding flags for
   everything to compile correctly (this corresponds to the ``-Q`` flag for Coq).
-  
-  As of today, we only support composition with libraries defined in the same
-  scope (i.e., under the same :ref:`dune-project` domain). This restriction will
-  be lifted in the future. Note that composition with the Coq standard library
-  is supported, but in this case the ``Coq`` prefix has been made available in a
-  qualified way, since :ref:`Coq lang 0.2<coq-lang>`.
+
+  You may also depend on theories belonging to another :ref:`dune-project` as long
+  as they share a common scope under another :ref:`dune-project` file or a
+  :ref:`dune-workspace` file.
+
+  Doing so can be a simple as placing a Coq project within the scope of another.
+  This process is termed *composition*. See the :ref:`inter-project
+  composition<example-inter-project-theory>` example. Inter-project composition
+  has been available since :ref:`Coq lang 0.4<coq-lang>`.
```

Comment:
Maybe we can extend a bit more here: "when putting two projects together under the same workspace, composition means in practice that build rules in one project will be able to depend with fine granularity on objects in another one. This can be very useful in practice, and given two projects `A`, `B`, with `B` depending on `A`, building a target in `B` won't trigger a full rebuild of `A`.

Reply:
I've added something similar.